### ISSUE TITLE
False positive for C0103: Constant name doesn't conform to ..

### ISSUE THREAD

**Issue by bje-** [PyCQA/pylint#7523]
### Bug description

```
"""Test module"""

total = 0

while True:
    total += 1
```

### Command used

```shell
pylint test.py
```

### Pylint output

```shell
************* Module test
test.py:3:0: C0103: Constant name "total" doesn't conform to '(([A-Z_][A-Z0-9_]*)|(__.*__))$' pattern (invalid-name)
```

### Expected behavior

'total' is not a constant in this example. Although Python does not formally have constants, I would expect Pylint to only consider 'total' to be a constant if it is never reassigned.

### Pylint version

```shell
pylint 2.15.3
astroid 2.12.10
Python 3.10.6 (main, Aug  2 2022, 15:11:28) [GCC 9.4.0]
```

### OS / Environment

Ubuntu 22.04

**Comment by mbyrnepr2** [comment]
Thanks for reporting.
It's a duplicate of https://github.com/PyCQA/pylint/issues/7232.

I think we should stop emitting this invalid name since we don't know if it will be treated as a constant or not.

**Comment by bje-** [comment]
Don't we have data flow analysis to tell us if it's ever re-assigned? It's otherwise a useful warning.

**Comment by Pierre-Sassoulas** [comment]
@bje- could you check the discussion in #7232, please ? It's quite lengthy and exhaustive :)

**Comment by pan64** [comment]
additionally
```
var_name = "value" # will raise this C0103 message, but
var_name: str = "value # is accepted.
```
